libraries/CurieBLE/src/internal/BLECharacteristicImp.h

```diff
@@ -324,7 +324,7 @@ class BLECharacteristicImp: public BLEAttribute{
     bt_gatt_subscribe_params_t _sub_params;
     bool        _subscribed;
     
-    bool _reading;
+    volatile bool _reading;
```

Comment:
OK, so *why* do you need to force optimization off? Who is accessing this variable out of normal execution flow? Describe the situation that will occur if `_reading` is not `volatile`.